cmake/modules/RootConfiguration.cmake

```diff
@@ -636,6 +636,12 @@ else()
    set(has_found_attribute_always_inline undef)
 endif()
 
+if ("cxx_variable_templates" IN_LIST CMAKE_CXX_COMPILE_FEATURES)
```

Comment:
This just lists the features the compiler supports, but the compiler may need extra options to enable it, and that's what you are not handling here (e.g. adding C++14 flags to enable variable templates).

Reply:
yes, see my last comment...I'm not sure how to add the check for "c++ standard greater than 14" though. I could check for `cxxversion == cxx14 || cxxversion == cxx17` but that will break as soon as someone uses cxx20.

Reply:
You could probably try this: https://cmake.org/cmake/help/latest/command/try_compile.html